### DISCUSSION TITLE
toJSON returns ID of OneToOne relation

### DISCUSSION THREAD

**Discussion by ogdans3** [mikro-orm/mikro-orm#1184] [Q&A] [answered]
Im having some trouble with the OneToOne relation. Using the toJSON returns the ID of the B instance instead of the data. Printing the instance shows that all data is loaded, but it seems to be stripped when converting to JSON.

Is this the intended behaviour? 
Do I have to define a custom toJSON method to force the contents of the inverse side to be populated in the JSON?

```
@Entity()
export class A {
    @PrimaryKey()
    id: string = nanoid(40);

    @OneToOne({
        entity: () => B,
        joinColumn: "bId",
        nullable: true,
    })
    b?: B;
}

@Entity()
export class B {
    @PrimaryKey()
    id: string = nanoid(40);

    @Property()
    name = "1";
}

let testInitialize = async () => {
    const orm = await MikroORM.init({
        type: "sqlite",
        dbName: "database/test_mikroorm_db.sqlite",
        metadataProvider: TsMorphMetadataProvider,
        entities: [A, B],
        debug: false,
    });

    await schema(orm);

    let a = new A();
    a.b = new B();
    await orm.em.persistAndFlush(a);

    let storedA = await orm.em.findOne(A, a.id);
    console.log(storedA);
    console.log(storedA.toJSON());
};
```

What i wish to accomplish:
I want the call to     console.log(storedA.toJSON());
 to return a object like
```
{
    id: "some-id"
    b: {
        id: "some-other-id"
        name: "1"   
    }
}
```

I have tried to find out if this is the intended result or if there is something else I have done wrong.

**Comment by B4nan** [comment, marked as answer]
Yes, this is intended behaviour, you need to mark the property as populated (it happens automatically if you fetch an entity with populate hint, but here you are persisting new entity). 

Not sure why are you calling that `findOne(A, a.id)`, that will give you the very same entity as you already have in variable `a` (unless it happens in different context).

```ts
let a = new A();
a.b = new B();
await orm.em.persistAndFlush(a);
wrap(a.b).populated();
console.log(a.toJSON());
```

Or with the populate hint:

```ts
let storedA = await orm.em.findOne(A, a.id, ['b']); // or via FindOptions: `{ populate: ['b'] }`
console.log(storedA.toJSON());
```

Or you could enable this behaviour for newly persisted entities automatically via `populateAfterFlush: true` in the ORM options.

**Comment by ogdans3** [comment]
I see. I did not know about the ```wrap(a.b).populated();```. It works perfectly, thank you 👍 

Great project btw
